src/components/Api.js

```diff
@@ -0,0 +1,104 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._url = baseUrl;
+    this._header = headers;
+  }
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`);
+  }
+
+  getInitialCards() {
+    return fetch(`${this._url}/cards`, { headers: this._header }).then((res) =>
+      this._checkResponse(res)
+    );
+  }
+
+  getUserInfo() {
+    return fetch(`${this._url}/users/me`, { headers: this._header })
+      .then((res) => this._checkResponse(res))
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+
+  updateUserProfile({ name, about }) {
+    return fetch(`${this._url}/users/me`, {
+      method: "PATCH",
+      headers: this._header,
+      body: JSON.stringify({
+        name: `${name}`,
+        about: `${about}`,
+      }),
+    })
+      .then((res) => this._checkResponse(res))
+      .catch((err) => {
```

Comment:
Needs correction: Errors should be handled outside of the api class in `index.js`. Now `catch` block is placed in the middle of the chain, so the rest of the chain won't know anything about the error — but it should because we need to update the data only in case of success

The api call in `index.js` should look like this:

```js
// show loading
api.
  .then(...)
  .catch(...)
  .finally(() => {
    // hide loading
  })
```